### ISSUE TITLE
Model caching option..

### ISSUE THREAD

**Issue by shez1983** [laravel/ideas#1616]
so currently if i need caching, i resort to packages - and although they do an ok job none so far work with relationship ie if you did 

```User::with('friends')...```

it may be cached but not DELETED if friends are modified. so its a pain currently (unless I am missing something) 

either way an inbuilt caching for model which can be toggled with using .ENV setting (kind of like paginate) or a property on model $caching = true;  (kind of like when you specify with=[], casts[]... in the model)

**Comment by alexbowers** [comment]
Automatic caching is a very difficult thing to achieve.

If there is an update query, or any query that does not take place through Eloquent, then it would be impossible to invalidate the cache correctly.

You then have other issues, if you have a query for search results, you then store a new record in the database, you need to figure out should that search be updated. This again is very difficult to solve.

It is better for the framework to leave it to the  application developers themselves to figure out what needs caching, and under what conditions it needs to be cached.

**Comment by ollieread** [comment]
It very much sounds like what you want is a sort of entity manager, though you're essentially creating a data layer on top of the already existing data layer. If you truly need this and genuinely do think it would benefit you and others, I'd suggest releasing it as a package.

**Comment by shez1983** [comment]
@alexbowers 

so then people who have made caching packages are 'stupid' for trying to create a package? i mean if what you said is true there shouldnt be any caching

**Comment by alexbowers** [comment]
Not at all, it is a problem that people often try to solve, but there is no one-fix solution for it, so the framework should stay out of that space. 

If third-party packages want to try solving the problem, they can do, but the framework should not.

**Comment by shez1983** [comment]
this doesnt make sense - caching is something that CAN be added. there can be a solution for everyone a basic mechanism that caches models/relationships and delets them. anything else user will have to do themselves.

**Comment by alexbowers** [comment]
How should the framework handle `update` queries?

These aren't trackable by Eloquent to automatically clear the cache for those records, nor for the caches for any searches that would now involve, or no longer involve, those records.
